build/wiki.ysh

```diff
@@ -77,20 +89,73 @@ proc render-one(path) {
         <head>
           <meta name="viewport" content="width=device-width, initial-scale=1">
           <title>$title</title>
+
           <link rel="stylesheet" type="text/css" href="../web/base.css" />
+
+          <style>
+            #search > div {
+              margin-top: 8px;
+              width: 100%;
+            }
+
+            #search input[type=text], #search button {
+              padding: 4px 8px;
+
+              background-color: #EEE;
+              color: #444;
+            }
+
+            /* We cannot control the search UI, so manually calculate widths for sizing */
+            #search input {
+              padding-right: 4px !important; /* Pagefind likes to add a large padding here, disable it */
+
+              width: calc(100%
+                            - max(10%, 50px) /* button width */
+                            - 20px /* margins */);
+
+              border: 1px solid #EEE;
+            }
+
+            #search button {
+              width: calc(max(10%, 50px));
+
+              border: 1px solid #AAA;
+            }
+
+            /* This is the load more results button */
+            #search .pagefind-ui__drawer button {
+              margin-left: 0;
+              margin-right: 0;
+              width: 100%;
+            }
+          </style>
         </head>
         <body class="width40">
+
+        <div id="search"></div>
+        <hr>
     """
 
     pre-render-wikilinks <$path | cmark
 
     echo """
+        <script src="./pagefind/pagefind-ui.js"></script>
```

Comment:
does this file need to be added to github pages?

The CSS could als go in a separate file perhaps ... it will make each page smaller